src/events.rs

```diff
@@ -119,6 +135,17 @@ impl<T: System> EventsDecoder<T> {
     {
         let size = U::default().encode().len();
         self.type_sizes.insert(name.to_string(), size);
+        // A segmenter decodes a type from an input stream (&mut &[u8]) and returns the serialized
+        // type to the output stream (&mut Vec<u8>).
+        self.type_segmenters.insert(
```

Comment:
Looks like `type_sizes` can be removed now

Reply:
Removed. However I didn't change the public function names `check_missing_type_sizes` and `register_type_size` because we don't want to break the downstream code.